### PR TITLE
feat: New item button for crud

### LINKED ISSUE #1420: [autocrud] Add a button for creating a new entity
There should be a "New" button that enables an empty form so you can add a new item

### PR DESCRIPTION
**Pull request by Artur-**
Fixes #1420

### CODE REVIEW

File: packages/ts/react-grid/test/autocrud.spec.tsx [hunk 1]
```diff
@@ -131,6 +131,61 @@ describe('@hilla/react-grid', () => {
...
+      await submit(result);
+      await nextFrame();
+      await nextFrame();
+      await nextFrame();
+      await setFormField(result, 'First name', 'Jerp');
+      await submit(result);
+      await nextFrame();
+      await nextFrame();
+      await nextFrame();
+      expect(getBodyCellContent(grid, 1, 0).innerText).to.equal('Jerp');
```
Comment (sissbruecker) [inline packages/ts/react-grid/test/autocrud.spec.tsx:187]: Could also assert that the number of items / rows doesn't increase.

File: packages/ts/react-grid/test/test-models-and-services.ts [hunk 2]
```diff
@@ -216,8 +216,13 @@ export const createService = <T extends HasIdVersion>(initialData: T[]): CrudSer
       const newValue = { ...value };
       if (currentValue) {
         newValue.version = currentValue.version + 1;
+        data = data.map((item) => (item.id === newValue.id ? newValue : item));
+      } else {
+        // newValue.id =
+        newValue.id = data.map((item) => item.id).reduce((prev, curr) => Math.max(prev, curr)) + 1;
```
Comment (sissbruecker) [inline packages/ts/react-grid/test/test-models-and-services.ts:222]:
```suggestion
        newValue.id = data.map((item) => item.id).reduce((prev, curr) => Math.max(prev, curr)) + 1;
```

File: packages/ts/react-grid/src/autocrud.tsx [hunk 3]
```diff
@@ -31,24 +33,31 @@ export function ExperimentalAutoCrud<TItem>({ service, model, noDelete }: AutoCr
...
-              if (e.detail.value.length === 0) {
-                setItem(undefined);
-              } else {
-                const selectedItem = e.detail.value[0];
-                setItem({ ...selectedItem });
-              }
-            }}
-            customColumns={customColumns}
-          ></AutoGrid>
+          <VerticalLayout style={{ flex: 1 }}>
```
[the remark this answers is not in the document]
Comment (Artur-) [reply packages/ts/react-grid/src/autocrud.tsx:36]: Then the new button would be placed over the form?

[the remark this answers is not in the document]
Comment (Artur-) [reply packages/ts/react-grid/src/autocrud.tsx:36]: The inspiration came from here https://v-herd.eu/jonte-vaadinplus/grids

File: packages/ts/react-grid/test/autocrud.spec.tsx (hunk 1 above)
Comment (Artur-) [reply packages/ts/react-grid/test/autocrud.spec.tsx:187]: Added

File: packages/ts/react-grid/src/autocrud.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sissbruecker) [reply packages/ts/react-grid/src/autocrud.tsx:36]: Looks weird if the form is not aligned with the grid. Also if you want to add a header / title to the CRUD that should also be above the form.

**Review by sissbruecker — APPROVED** [review]

File: packages/ts/react-grid/src/autocrud.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Artur-) [reply packages/ts/react-grid/src/autocrud.tsx]: Swapped the layouts

**Pull request merged**